### PR TITLE
Refactor Tests

### PR DESCRIPTION
**Pull request by tom-downes-and**
- Refactor Quiz test

### CODE REVIEW

**Comment by tom-downes-and** [comment]
- Removed async act render - this isn't necessary
- Added repetition of actions within each test. Whilst DRY principles generally apply, in tests it is often clearer to repeat the same lines from test to test. This can aid in clarity when others are reading the tests.
- Used jest.each for the quiz score test

File: src/components/games/quiz/Quiz.test.tsx [hunk 1]
```diff
@@ -23,317 +16,282 @@ const mockContent = {
...
+  ]
 };
-let user : UserEvent;
-
-beforeEach(() => {
-  jest.spyOn(FirebaseAPI, "fetchQuizData").mockImplementation(() => Promise.resolve(mockContent));
-});
-
-afterEach(() => {
-  cleanup();
```
Comment (tom-downes-and) [inline src/components/games/quiz/Quiz.test.tsx:211]: Don't need to do a cleanup

File: src/components/games/quiz/Quiz.test.tsx [hunk 2]
```diff
@@ -23,317 +16,282 @@ const mockContent = {
...
-    const begin = screen.getByTestId("game-begin");
-    expect(begin).toBeInTheDocument();
-  });
-
   describe("Quiz functionality", () => {
     beforeEach(async () => {
-      user = userEvent.setup();
-      const quizRoute = "/Games/Quiz";
-
-      await act(async () => {
```
Comment (tom-downes-and) [inline src/components/games/quiz/Quiz.test.tsx:268]: This is a bit wacky!

File: src/components/games/quiz/Quiz.test.tsx [hunk 3]
```diff
@@ -23,317 +16,282 @@ const mockContent = {
...
-        await render(
-          <RoutingTestWrapper path={quizRoute}>
```
Comment (tom-downes-and) [inline src/components/games/quiz/Quiz.test.tsx:270]: No need to wrap in a route, we are only rendering one component

File: src/components/games/quiz/Quiz.test.tsx [hunk 4]
```diff
@@ -23,317 +16,282 @@ const mockContent = {
...
+          </QuizContext>
+        </GameContext>,
+        { wrapper: MemoryRouter }
+      );
 
-    test("should alert whether answer was correct", async() => {
-      await clickCorrectAnswer();
-      expect(screen.getByText("Well done, you got the right answer!")).toBeInTheDocument();
-      await clickNext();
+      await user.click(screen.getByRole("button", { name: /begin/i }));
```
Comment (tom-downes-and) [inline src/components/games/quiz/Quiz.test.tsx:218]: Tried to use roles as much as possible - accessibility first

File: src/components/games/quiz/Quiz.test.tsx [hunk 5]
```diff
@@ -23,317 +16,282 @@ const mockContent = {
...
 
-    test("should score 10 out of 10", async() => {
-      await playQuizWithScore(10);
+      await user.click(screen.getByRole("button", { name: /^correct answer/i }));
+      expect(screen.getByText(/1 out of 10/)).toBeVisible();
+    });
 
-      const quizScore = screen.getByTestId("modal-title");
-      expect(quizScore).toHaveTextContent("Score: 10 out of 10");
+    test.each([
```
Comment (tom-downes-and) [inline src/components/games/quiz/Quiz.test.tsx:266]: Repeat test for each set of params - syntax is a bit weird though

File: src/components/games/quiz/Quiz.test.tsx [hunk 6]
```diff
@@ -23,317 +16,282 @@ const mockContent = {
...
+        .mockImplementation(() => Promise.resolve(mockContent));
     });
 
-    test("First question renders", () => {
-      expect(screen.getByText("First question")).toBeInTheDocument();
-    });
+    test("should alert whether answer was correct", async () => {
+      const user = userEvent.setup();
+
+      render(
```
Comment (tom-downes-and) [inline src/components/games/quiz/Quiz.test.tsx:209]: This is repeated in every test so could go in a beforeEach, but I think it is nice to be explicit in each test what is being rendered.

File: src/components/games/quiz/Quiz.test.tsx (hunk 5 above)
Comment (AshleyBeel) [reply src/components/games/quiz/Quiz.test.tsx:266]: I've never come across this syntax before, is there any reason why strings can't be used? E.g:

`expect(screen.getByText("better luck next time")).toBeVisible();`

Comment (tom-downes-and) [reply src/components/games/quiz/Quiz.test.tsx:266]: Strings can be used and normally I would use strings myself. Can't really explain why I chose to use regex this time!

**Review by leahshields-and — APPROVED** [review]

**Pull request merged**